core/api/rpc/make.hpp

```diff
@@ -48,50 +48,50 @@ namespace fc::api {
             return respond(Response::Error{
                 kInvalidParams, errorToPrettyString(maybe_params.error())});
           }
-          auto maybe_result = std::apply(method, maybe_params.value());
-          if (!maybe_result) {
-            return respond(Response::Error{
-                kInternalError, errorToPrettyString(maybe_result.error())});
-          }
-          if constexpr (!std::is_same_v<Result, void>) {
-            auto &result = maybe_result.value();
-            if constexpr (is_chan<Result>{}) {
-              result.id = make_chan();
-            }
-            if constexpr (is_wait<Result>{}) {
-              result.waitOwn([respond{std::move(respond)}](auto maybe_result) {
-                if (!maybe_result) {
-                  respond(Response::Error{
-                      kInternalError,
-                      errorToPrettyString(maybe_result.error())});
-                } else {
-                  respond(encode(maybe_result.value()));
-                }
-              });
-              return;
-            } else {
-              respond(api::encode(result));
-            }
-            if constexpr (is_chan<Result>{}) {
-              result.channel->read(
-                  [send{std::move(send)}, chan{result}](auto opt) {
-                    if (opt) {
-                      send(kRpcChVal,
-                           encode(std::make_tuple(chan.id, std::move(*opt))),
-                           [chan](auto ok) {
-                             if (!ok) {
-                               chan.channel->closeRead();
-                             }
-                           });
-                    } else {
-                      send(kRpcChClose, encode(std::make_tuple(chan.id)), {});
-                    }
-                    return true;
-                  });
+
+          if constexpr (is_chan<Result>{}) {
```

Comment:
move channel processing to callback, when method is called.
will block if method is proxy and calls other method through rpc